macros/date.njk

```diff
@@ -14,15 +14,15 @@
     <div class="form-date" id="{{ name }}[label]">
       <div class="form-group form-group-day">
         <label class="form-label" id="{{ name }}[day][label]" for="{{ name }}[day]">{{ t('Day') }}</label>
-        <input class="form-control {% if dayError or dateError %} form-control-error {% endif %}" id="{{ name }}[day]" name="{{ name }}[day]" value="{{ form.valueFor(name).day }}" type="number" aria-describedby="dob-hint">
+        <input class="form-control {% if dayError or dateError %} form-control-error {% endif %}" id="{{ name }}[day]" name="{{ name }}[day]" value="{{ form.valueFor(name).day | default('') }}" type="number" aria-describedby="dob-hint">
```

Comment:
when does this happen?

Reply:
When either `form.valueFor(name)` or `form.valueFor(name).day` are undefined. This is due to the following custom deserialisation:

```
const instance = new LocalDate();
    ['year', 'month', 'day'].forEach((field: string) => {
      if (value[field]) {
        instance[field] = _.toNumber(value[field])
      }
    })
    return instance
```

That avoids 0 to be displayed on the form when user did not provide value.

Reply:
when does that happen?
![image](https://user-images.githubusercontent.com/21194782/31266311-12efb940-aa6b-11e7-8b9c-20996baa1b1c.png)